lib/Node.php

```diff
@@ -11,6 +11,8 @@
  * @author Evert Pot (http://evertpot.com/)
  * @license http://sabre.io/license/ Modified BSD License
  */
+
+/** @implements \IteratorAggregate<int, Node> */
```

Comment:
I am not too sure if this is the correct approach ... it can also hold sub classes of Node .....